internal/config/config.go

```diff
@@ -118,12 +118,55 @@ type RoutingConfig struct {
 	MaxBlocksBehind int `yaml:"maxBlocksBehind"`
 }
 
-type Config struct {
+type SingleChainConfig struct {
+	ChainName string `yaml:"chainName"`
 	Upstreams []UpstreamConfig
 	Groups    []GroupConfig
-	Global    GlobalConfig
 	Routing   RoutingConfig
-	ChainName string `yaml:"chainName"`
+}
+
+func (c *SingleChainConfig) isValid() bool {
+	isChainConfigValid := true
+	isChainConfigValid = isChainConfigValid && IsGroupsValid(c.Groups)
+
+	for _, upstream := range c.Upstreams {
+		isChainConfigValid = isChainConfigValid && upstream.isValid(c.Groups)
+	}
+
+	if c.ChainName == "" {
+		zap.L().Error("chainName cannot be empty", zap.Any("config", c))
+
+		isChainConfigValid = false
+	}
+
+	return isChainConfigValid
+}
+
+func isChainsValid(chainsConfig []SingleChainConfig) bool {
+	isValid := len(chainsConfig) > 0
+
+	for chainIndex := range chainsConfig {
```

Comment:
The reason I started using this pattern was to avoid copying the slice objects by value in case they're large. I even stumbled upon some discussions where folks recommended sticking to this pattern consistently so I've been leaning towards a bit more.

Reply:
Oh I see. Makes sense!